Adyen/Webhooks/BalancePlatformWebhookHandler.cs

```diff
@@ -0,0 +1,187 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.Serialization;
+using Adyen.Model.ReportNotification;
+using Adyen.Model.ConfigurationNotification;
+using Adyen.Model.TransferNotification;
+using Adyen.Util;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Adyen.Webhooks
+{
+    public class BalancePlatformWebhookHandler
+    {
+        /// <summary>
+        /// Deserializes to a generic banking webhook from the <paramref name="jsonPayload"/>. Use this either to catch the
+        /// webhook type if unknown or if the explicit type is not required.
+        /// </summary>
+        /// <param name="jsonPayload">The json payload of the webhook.</param>
+        /// <returns>The parsed webhook packaged in a dynamic object.</returns>
+        /// <exception cref="JsonReaderException">Throws when json is invalid.</exception>
+        public dynamic GetGenericBalancePlatformWebhook(string jsonPayload)
+        {
+            if (GetAccountHolderNotificationRequest(jsonPayload, out var accountHolderNotificationRequest)) 
```

Comment:
From integration perspective:

If the developer has to switch-case over all the notificationRequest types, we shouldn't provide the `GetGenericBalancePlatformWebhook` here as the logic will happen twice now from the integration-side

From library-side:
We should provide the developers a way to deserialize, then they can switch-case over the cases

Concrete example in java:
```java
@PostMapping("/webhooks/notifications")
    public ResponseEntity<String> webhooks(@RequestHeader Map<String, String> headers, @RequestBody String json) throws Exception {
        logger.info("/webhooks/notifications");

        EventHandler eventHandler = new EventHandler();
        BankingWebhookHandler webhookHandler = new BankingWebhookHandler(json);


        String hmacsignature = headers.get("hmacsignature");

        if (hmacsignature == null || hmacsignature.isBlank()) {
            logger.warn("HMAC Signature not found");
            throw new RuntimeException("HMAC Signature not found");
        }

        if (!hmacValidator.validateHMAC(json, hmacsignature, applicationProperty.getHmacKey())) {
            logger.warn("Invalid HMAC signature");
            throw new RuntimeException("Invalid HMAC signature");
        }

        String type = eventHandler.getEventType(json);
        String environment = eventHandler.getEventEnvironment(json);

        logger.info("Event " + type + " on " + environment);

        switch (type) {
            case "balancePlatform.accountHolder.created":
                webhookHandler.getAccountHolderNotificationRequest().ifPresent((AccountHolderNotificationRequest event) -> {
                    accountHolderService.createAccoundHolder(event.getData());
                });
                break;
            case "balancePlatform.accountHolder.updated":
                webhookHandler.getAccountHolderNotificationRequest().ifPresent((AccountHolderNotificationRequest event) -> {
                    accountHolderService.updateAccoundHolder(event.getData());
                });
                break;
            case "balancePlatform.balanceAccount.created":
                webhookHandler.getBalanceAccountNotificationRequest().ifPresent((BalanceAccountNotificationRequest event) -> {
                    balanceAccountService.createBalanceAccount(event.getData());
                });
                break;
            case "balancePlatform.balanceAccount.updated":
                webhookHandler.getBalanceAccountNotificationRequest().ifPresent((BalanceAccountNotificationRequest event) -> {
                    balanceAccountService.updateBalanceAccount(event.getData());
                });
                break;
            case "balancePlatform.paymentInstrument.created":
                webhookHandler.getPaymentNotificationRequest().ifPresent((PaymentNotificationRequest event) -> {
                    paymentInstrumentService.createPaymentInstrument(event.getData());
                });
                break;
            case "balancePlatform.paymentInstrument.updated":
                webhookHandler.getPaymentNotificationRequest().ifPresent((PaymentNotificationRequest event) -> {
                    paymentInstrumentService.updatePaymentInstrument(event.getData());
                });
                break;
            case "balancePlatform.balanceAccountSweep.created":
                webhookHandler.getSweepConfigurationNotificationRequest().ifPresent((SweepConfigurationNotificationRequest event) -> {
                    sweepConfigurationService.createSweepConfiguration(event.getData());
                });
                break;
            case "balancePlatform.balanceAccountSweep.updated":
                webhookHandler.getSweepConfigurationNotificationRequest().ifPresent((SweepConfigurationNotificationRequest event) -> {
                    sweepConfigurationService.updateSweepConfiguration(event.getData());
                });
                break;
            case "balancePlatform.balanceAccountSweep.deleted":
                webhookHandler.getSweepConfigurationNotificationRequest().ifPresent((SweepConfigurationNotificationRequest event) -> {
                    sweepConfigurationService.deleteSweepConfiguration(event.getData());
                });
                break;
            case "balancePlatform.transfer.created":
                webhookHandler.getTransferNotificationRequest().ifPresent((TransferNotificationRequest event) -> {
                    transferService.createTransfer(event.getData());
                });
                break;
            case "balancePlatform.transfer.updated":
                webhookHandler.getTransferNotificationRequest().ifPresent((TransferNotificationRequest event) -> {
                    transferService.updateTransfer(event.getData());
                });
                break;
            case "balancePlatform.report.created":
                webhookHandler.getReportNotificationRequest().ifPresent((ReportNotificationRequest event) -> {
                    reportService.getGeneratedReport(event.getData());
                });
                break;
            default:
                // deal with unexpected event (ie there is a new event that must be processed?)
                logger.error("Unexpected event type: " + type);
                throw new InvalidWebhookTypeException("Unexpected event type: " + type);
        }

        // Acknowledge event has been consumed
        return ResponseEntity.ok().body("[accepted]");
    }

```